manifests/interface.pp

```diff
@@ -46,17 +55,58 @@
     'CONNECTED_MODE' => $connected_mode,
   }
 
-  network::interface { $name:
-    ensure               => $ensure,
-    enable               => $enable,
-    onboot               => $onboot,
-    type                 => 'InfiniBand',
-    ipaddress            => $ipaddr,
-    netmask              => $netmask,
-    gateway              => $gateway,
-    nm_controlled        => 'no',
-    mtu                  => $mtu,
-    options_extra_redhat => $options_extra_redhat,
+  if $bonding {
+    if empty($bonding_slaves) {
+      fail("No slave interfaces given for bonding interface ${name}")
+    }
+
+    # Setup interfaces for the slaves
+    $bonding_slaves.each |String $ifname| {
+      if !($ifname in $facts['infiniband_netdevs']) {
```

Comment:
My only concern with this specific fact check is that if you run Puppet on a newly provisioned system for first time this will fail because it could be that infiniband stack isn't installed so command to get this fact won't exist. For a newly provisioned system it could take 2 puppet runs to fully apply the interfaces.

Reply:
Well, this check is actually only a guard clause to bail out if the admin is using a device that is not available. We can leave out this check entirely and rely on the admin using the correct interface names.

Reply:
I think leaving the check off would be best.  I was thinking maybe check the `infiniband_netdevs` fact is present and using a warning but it would be reasonable for a site to want to deploy ifcfg files that don't map to existing interfaces.  For example, in my environment we use NFS root where a read-write host is responsible for setting up files and sometimes we need that host to create files for things not necessarily present on itself but that will be present on compute nodes which have read-only access to the root filesystem.  Same would be true if someone is using Puppet to deploy files to a system that generates images, which I don't have experience with but can imagine is something done for compute environments.

Reply:
Very valid use cases where you want the ifcfg files being generated without actually having the interfaces. Ok, I'll remove the condition and the respective test case. Give me 5m.